src/OSVRTrackedController.cpp

```diff
@@ -0,0 +1,515 @@
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_ModeLabel_String, "");
+    //vr::VRProperties()->SetBoolProperty(propertyContainer_, vr::Prop_WillDriftInYaw_Bool, true);
+    //vr::VRProperties()->SetBoolProperty(propertyContainer_, vr::Prop_DeviceIsWireless_Bool, false);
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_DeviceClass_Int32,       static_cast<int32_t>(deviceClass_));
+    /*
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis0Type_Int32,         static_cast<int32_t>(analogInterface_[0].axisType));
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis1Type_Int32,         static_cast<int32_t>(analogInterface_[1].axisType));
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis2Type_Int32,         static_cast<int32_t>(analogInterface_[2].axisType));
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis3Type_Int32,         static_cast<int32_t>(analogInterface_[3].axisType));
+    vr::VRProperties()->SetInt32Property(propertyContainer_,  vr::Prop_Axis4Type_Int32,         static_cast<int32_t>(analogInterface_[4].axisType));
+    vr::VRProperties()->SetUint64Property(propertyContainer_, vr::Prop_SupportedButtons_Uint64, static_cast<int32_t>(NUM_BUTTONS));
+    */
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_ModelNumber_String,      "OSVR Controller");
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_SerialNumber_String,     name_.c_str());
+    // set the vive controller at the default.
+    vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_RenderModelName_String,  "vr_controller_vive_1_5");
+    //vr::VRProperties()->SetStringProperty(propertyContainer_, vr::Prop_RenderModelName_String, settings_->getSetting<std::string>("cameraRenderModel", "").c_str());
+}
+
+/**
+ * Registers a button based on the path and id.
+ * @param id the id of the button which is used for indexing an array.
+ * @param path the complete path to the button.
+ */
+void OSVRTrackedController::registerButton(int id, std::string path, vr::EVRButtonId button_id){
+    buttonInterface_[id].buttonInterface = context_.getInterface(path);
+    if (buttonInterface_[id].buttonInterface.notEmpty()) {
+	buttonInterface_[id].button_id        = button_id;
+	buttonInterface_[id].parentController = this;
+	buttonInterface_[id].buttonInterface.registerCallback( &OSVRTrackedController::controllerButtonCallback, &buttonInterface_[id]);
+    }else{
+	buttonInterface_[id].buttonInterface.free();
+    } 
+}
+
+/**
+ * Registers a touch button based on the path and id.
+ * @param id the id of the button which is used for indexing an array.
+ * @param path the complete path to the button.
+ */
+void OSVRTrackedController::registerButtonTouch(int id, std::string path, vr::EVRButtonId button_id){
```

Comment:
Probably, I had a go at it but I am not skilled enough in C++ to do this.  Sorry.